llama-index-integrations/embeddings/llama-index-embeddings-oracleai/llama_index/embeddings/oracleai/base.py

```diff
@@ -0,0 +1,167 @@
+# -----------------------------------------------------------------------------
+# Authors:
+#   Harichandan Roy (hroy)
+#   David Jiang (ddjiang)
+#
+# -----------------------------------------------------------------------------
+# ...embeddings/oracleai.py
+# -----------------------------------------------------------------------------
+
+from __future__ import annotations
+
+from typing import Any, Dict, List, Optional, TYPE_CHECKING
+
+import json
+
+from llama_index.core.bridge.pydantic import PrivateAttr
+from llama_index.core.embeddings import BaseEmbedding
+
+if TYPE_CHECKING:
+    from oracledb import Connection
+
+"""OracleEmbeddings class"""
+
+
+class OracleEmbeddings(BaseEmbedding):
+    """Get Embeddings."""
+
+    _conn: Any = PrivateAttr()
+    _params: Dict[str, Any] = PrivateAttr()
+    _proxy: Optional[str] = PrivateAttr()
+
+    def __init__(
+        self,
+        conn: Connection,
+        params: Dict[str, Any],
+        proxy: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        super().__init__(**kwargs)
+        self._conn = conn
+        self._proxy = proxy
+        self._params = params
+
+    @classmethod
+    def class_name(self) -> str:
+        return "OracleEmbeddings"
+
+    """
+    1 - user needs to have create procedure, create mining model, create any directory privilege.
+    2 - grant create procedure, create mining model, create any directory to <user>;
+    """
```

Comment:
Can we remove this random string?

Reply:
related comment, let me move to the right place.